Problem: Feature checking inconsistencies (#416)
zeromq/cppzmq: README.md

}
```

Compatibility Guidelines
========================

> The users if cppzmq are expected follow the guidelines below for breakage-free behaviour when upgrading cppzmq to newer versions (non-exhaustive list):

Reviewer: Some editorial improvements:
```suggestion
The users of cppzmq are expected to follow the guidelines below to ensure not to break when upgrading cppzmq to newer versions (non-exhaustive list):
```

> * Do not depend on any macros defined in cppzmq unless explicitly exported.

Reviewer: I would use a different term than `exported` here. Under `exported`, I'd expect some technical mechanism, but you simply mean that it is declared as such in the readme below.

What about saying:
```suggestion
* Do not depend on any macros defined in cppzmq unless explicitly declared public here.
```

> The following macros are exported: `CPPZMQ_VERSION`, `CPPZMQ_VERSION_MAJOR`, `CPPZMQ_VERSION_MINOR`, `CPPZMQ_VERSION_PATCH`. The following macros may be defined to opt in/out of behaviour:

Reviewer: These are two distinct aspects, and should be separated better. One is lead to think that the second sentence is related to exported/public macros as well, but on the contrary, it is related to macros that are imported/consumed. This should be made clearer.
